src/main/scala/battleships/Grid.scala

```diff
@@ -0,0 +1,57 @@
+package battleships
+
+import battleships.Ship.Ship
+
+class Grid(s: Int) {
+  val grid = Array.ofDim[Ship](s, s)
+  val size = s
+
+  var lastx: Int = -1
+  var lasty: Int = -1
+
+  def printGrid(): Unit = {
+    var s: Ship = null
+    System.out.println()
+    for (i <- 0 to size) {
+      if (i == 0) {
+        print("  ")
+        for (k <- 0 to size) {
+          print(s"$k ")
+        }
+        println()
+      }
+
+      for (j <- 0 to size) {
+        s = grid(i)(j)
+        if (j == 0) print(s"$i ")
+        if (i == lasty && j == lastx) print("X ")
+        if (s == null)                print("~ ")
```

Comment:
all these checks for "s" could be a helper method to reduce the size of the current one and make it more readable